src/psyclone/psyir/tools/dependency_tools.py

```diff
@@ -164,39 +164,51 @@ def is_array_parallelisable(self, loop_variable, var_info):
         # a(i,j) and a(j+2, i-1) in one loop:
         # In this case the dimensions 1 (a(i,j)) and 0 (a(j+2,i-1)) would
         # be accessed. Since the variable is written somewhere (read-only
-        # was tested above), the variable can not be used in parallel.
+        # was tested above), the variable cannot be used in parallel.
         # Additionally, collect all indices that are actually used, since
         # they are needed in a test further down.
-        found_dimension_index = -1
+        found_dimension_index = (-1, -1)
```

Comment:
As suggested somewhere else, I think an initial value of `None` would be cleaner. Please could you also add comments describing what `found_dimension_index` and `all_indices` will hold.

Reply:
In fact, unless my review has got very confused, is this code substantially duplicating similar functionality elsewhere? If so, is there any way of reducing duplication?

Reply:
Indeed. There is a difference in my further work now, but I will check if I can either use the same code in both places and make this a standalone function (#1269). Or some other way to avoid this duplication.